presto-main/src/main/java/io/prestosql/server/security/AuthenticationFilter.java

```diff
@@ -43,12 +44,16 @@
 public class AuthenticationFilter
         implements Filter
 {
+    private static final String HTTPS_PROTOCOL = "https";
+
     private final List<Authenticator> authenticators;
+    private final SecurityConfig securityConfig;
```

Comment:
The config objects are not thread safe, so we don't hold on to them after the constructor.  Instead add a `private final boolean httpsForwardingEnabled` field.